lib/net/pool.js

```diff
@@ -204,16 +209,22 @@ class Pool extends EventEmitter {
    */
 
   resetChain() {
-    if (!this.options.checkpoints)
+    if (!this.options.checkpoints && !this.options.neutrino)
       return;
 
-    this.checkpoints = false;
+    if (!this.options.neutrino)
+      this.checkpoints = false;
     this.headerTip = null;
     this.headerChain.reset();
     this.headerNext = null;
 
     const tip = this.chain.tip;
-
+    if (this.options.neutrino) {
+      this.headerChain.push(new HeaderEntry(tip.hash, tip.height));
+      this.cfHeaderChain = new List();
+      this.cfHeaderChain.push(new CFHeaderEntry(consensus.ZERO_HASH, 0));
+      return;
+    }
```

Comment:
why do we need neutrino logic in this function?

Reply:
We don't need cfheaderchain if it is not neutrino mode.

Reply:
oh okay I see well maybe you can move this logic to the end of the function so you dont have to duplicate the `headerChain` line ?

Reply:
it also may be cleaner to separate this out into two functions `resetHeadersChain()` and `resetFilterHeaders()` or something